lms/services/digest.py

```diff
@@ -0,0 +1,189 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from functools import lru_cache
+from typing import Dict, List, NewType, Optional
+
+from lms.models import (
+    Assignment,
+    AssignmentGrouping,
+    AssignmentMembership,
+    Course,
+    Grouping,
+    LTIRole,
+    User,
+)
+from lms.services import h_api
+
+
+@dataclass
+class UserRecords:
+    """All models.User rows for a given h_userid."""
```

Comment:
Really struggling to come up with a good name for this class but the idea is that it's the concept of an `ApplicationInstance`-independent user object that knows about all the `ApplicationInstance`'d "user records" for a given `h_userid`